### PR TITLE
[CI] Add MKLML build and test

### PR DESCRIPTION
**Pull request by mli**
1. added mklml build and test
2. removed `tests/python/gpu/test_forward.py`, which will be covered in integration test.
3. used a new version pip instead of pip 1.5.4 shipped with ubuntu.

### CODE REVIEW

**Comment by mli** [comment]
@piiswrong please have a check

File: tests/ci_build/Dockerfile.mklml_gpu [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# the reason we used a gpu base container because we are going to test MKLDNN
+# operator implementation against GPU implementation
+
+COPY install/ubuntu_*.sh /install/
+
+RUN /install/ubuntu_install_core.sh
+RUN /install/ubuntu_install_python.sh
+RUN /install/ubuntu_install_scala.sh
+
+RUN wget --no-check-certificate -O /tmp/mklml.tgz https://github.com/dmlc/web-data/raw/master/mxnet/mklml-release/mklml_lnx_2017.0.2.20170209.tgz
```
Comment (piiswrong) [inline tests/ci_build/Dockerfile.mklml_gpu]: hard coding it maybe a problem. What should @zhenlinluo do when there is a new release?

File: tests/python/gpu/test_forward.py [hunk 2]
```diff
@@ -1,61 +0,0 @@
-import os
```
Comment (piiswrong) [inline tests/python/gpu/test_forward.py]: why remove this?

File: tests/ci_build/Dockerfile.mklml_gpu (hunk 1 above)
Comment (mli) [inline tests/ci_build/Dockerfile.mklml_gpu]: added a comment in `prepare_mkl.sh` that one should also update this dockfile if there is a new release

File: tests/python/gpu/test_forward.py (hunk 2 above)
Comment (mli) [inline tests/python/gpu/test_forward.py]: it's requires downloading data and scipy. will move the test into a following integration test.

Comment (piiswrong) [inline tests/python/gpu/test_forward.py]: unittests also need to download data right?

Comment (mli) [inline tests/python/gpu/test_forward.py]: that's true. but this test is somewhat duplicated with https://github.com/dmlc/mxnet/blob/master/example/image-classification/test_score.py

Comment (piiswrong) [inline tests/python/gpu/test_forward.py]: Its here to gurrentee numerical consistency between cpu and gpu implementation. Error may not be big enough to change acc.

**Comment by mli** [comment]
@piiswrong added back test_forward. should be ready to merge.

**Pull request merged**